docs/docker-stack/changelog.rst

```diff
@@ -34,14 +34,29 @@ the Airflow team.
        any Airflow version from the ``Airflow 2`` line. There is no guarantee that it works, but if it does,
        then you can use latest features from that image to build the previous Airflow versions.
 
-:warning: Some of the images below (as noted in the Changelog) have been regenerated using newer
-       ``Dockerfiles``. This happens when there is a breaking change that invalidates already released images
-       and the images need regeneration. This has happened already when MySQL changed the keys they
-       used to release their packages: `#20911 <https://github.com/apache/airflow/issues/20911>`_
-       and 2.1 images were all regenerated using the 2.2 ``Dockerfile``. This is a rare event and
-       we do it only when we have no choice because of external factors. In such case, the newer version of
-       the image **might** contain breaking changes when it comes to running or building the image (but
-       we try to avoid those).
+Changes after publishing the images
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Important note!
+
+Some of the images below might be sometimes regenerated using newer ``Dockerfiles`` or constraints.
+This happens when there is an issue found or breaking change released by some of our dependencies
+that invalidates already released image and the images need refreshing. While we cannot assure 100% backwards
+compatibility when it happens, we at least document it here so that users affect might find it out and
+be aware of the changes.
```

Comment:
```suggestion
Occasionally our images need to be regenerated using newer ``Dockerfiles`` or constraints.
This happens when an issue is found or a breaking change is released by our dependencies
that invalidates the already released image, and regenerating the image makes it usable again.
While we cannot assure 100% backwards compatibility when it happens, we at least document it
here so that users affected can find the reason for the changes.
```

nit: I reworded this paragraph, reads better to me.